### PR TITLE
[Test] Reduce Katib GitHub Action Runs

### PR DESCRIPTION
**Pull request by andreyvelich**
Related: https://github.com/kubeflow/katib/issues/2024.

I've made few changes to improve our Actions execution.

- Rename all e2e test files to make it easier to identify them.
- In this PR: https://github.com/kubeflow/katib/pull/1998, we disabled publish actions on Katib UI changes. I think, we should disable only e2e for the Katib UI changes. We want to give user an option to install all Katib Control Plane images with the particular GitHub commit. That is why we publish all Katib images on every PR.
- I removed `postgres` e2e from all test and created new e2e test: `e2e-test-ui-random-search-postgres.yaml`. In that test we check that Katib UI can be deployed on minikube + run random search with `postgres` DB.
- Move `shellcheck` and YAML lint to the 1 workflow. I think, in the future PRs we should combine all unit test/lint checks in the 1 workflow.
- Remove `push` execution from all unit/lint tests.
- Add `cancel-in-progress` flag to our e2e workflows.
- Use single job for Katib Charmed actions.

/assign @johnugeorge @tenzen-y @anencore94 @kimwnasptd @orfeas-k

### CODE REVIEW

**Comment by andreyvelich** [comment]
`cancel-in-progress` seems to be working for E2E: https://github.com/kubeflow/katib/actions/runs/3534409566
Thank you @terrytangyuan!

**Comment by andreyvelich** [comment]
@DomFleischmann @knkski @ca-scribner
Please can you take a look on Katib Charmed error, I didn't make any changes for the execution process:
```
+ juju bundle deploy --build --destructive-mode --serial
Building and deploying bundle from bundle.yaml
Building katib-controller
Skipping 'bases[0].build-on[0]': channel '20.04' does not match host '22.04'.
No suitable 'build-on' environment found in 'bases[0]' configuration.
No suitable 'build-on' environment found in any 'bases' configuration. (full execution logs in '/tmp/charmcraft-log-swxyqidp')
Error: Error running subcommand ``charmcraft pack -p ./katib-controller --destructive-mode``: `exit status: 1`
Error: Process completed with exit code 1.
```

**Review by anencore94 — APPROVED** [review]
Looks much clear ! Thanks for your efforts.
Just one question about `Remove push execution from all unit/lint tests.`

If there is no master branch protection rule such as `Require branches to be up to date before merging`, I think we need to check such tests after the PR is merged too. WDYT ?

**Review by terrytangyuan — APPROVED** [review]
Great!

**Comment by johnugeorge** [comment]
> @DomFleischmann @knkski @ca-scribner Please can you take a look on Katib Charmed error, I didn't make any changes for the execution process:
> 
> ```
> + juju bundle deploy --build --destructive-mode --serial
> Building and deploying bundle from bundle.yaml
> Building katib-controller
> Skipping 'bases[0].build-on[0]': channel '20.04' does not match host '22.04'.
> No suitable 'build-on' environment found in 'bases[0]' configuration.
> No suitable 'build-on' environment found in any 'bases' configuration. (full execution logs in '/tmp/charmcraft-log-swxyqidp')
> Error: Error running subcommand ``charmcraft pack -p ./katib-controller --destructive-mode``: `exit status: 1`
> Error: Process completed with exit code 1.
> ```

This error is seen in other PRs as well. Can you help fix it at the earliest? @DomFleischmann @knkski @ca-scribner

**Comment by johnugeorge** [comment]
I am not sure if we need to publish  all Katib Control Plane images during every commit.    We will have to reconsider as this will slow down the whole development process. The user can build images any time using https://github.com/kubeflow/katib/blob/master/scripts/v1beta1/build.sh 

Particularly for Katib UI changes,  there is no point in building suggestion algorithms or darts/enas/mnist

File: .github/workflows/test-node.yaml [hunk 1]
```diff
@@ -1,9 +1,12 @@
 name: Frontend Test
 
 on:
-  - push
   - pull_request
 
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
```
Comment (kimwnasptd) [inline .github/workflows/test-node.yaml:9]: @andreyvelich I'm not fully in sync with why we need to use `concurrency`. Could you provide some more context?

**Comment by DomFleischmann** [comment]
> This error is seen in other PRs as well. Can you help fix it at the earliest? @DomFleischmann @knkski @ca-scribner

This seems to be because the we are using `ubuntu-latest` [here](https://github.com/kubeflow/katib/blob/0a1cb313642a4fe6bdd72fd6031fca8cc0b43d6c/.github/workflows/test-charmed-katib.yaml#L35) which lately was updated to 22.04. By pinning this to 20.04 again things should work.

File: .github/workflows/test-node.yaml (hunk 1 above)
Comment (anencore94) [reply .github/workflows/test-node.yaml:9]: It is useful when the developer push multiple commits on an open PR in short period.
https://github.com/kubeflow/katib/issues/2024#issuecomment-1320524310

To cancel previous-running workflows and run with new source codes.

**Comment by andreyvelich** [comment]
> I am not sure if we need to publish all images during every commit. We will have to reconsider as this will slow down the whole development process. The user can build images any time using https://github.com/kubeflow/katib/blob/master/scripts/v1beta1/build.sh

@johnugeorge I don't think it will slow down the UI development. I disabled the most time consuming building suggestion images on pull requests: https://github.com/kubeflow/katib/blob/01f4bcd93596a2a4a34aeb38f41195496e294491/.github/workflows/publish-algorithm-images.yaml#L5-L7
Do you have any concerns to have these actions when PR is merged ?

I think, we have users who want to get the latest Katib features, and they can easily bump all of their images to the particular commit. If we don't push images of every commit, users have to check what commit they should take for each image.
WDYT @johnugeorge @gaocegege @tenzen-y ?

File: .github/workflows/test-node.yaml (hunk 1 above)
Comment (andreyvelich) [reply .github/workflows/test-node.yaml:9]: Yes, @anencore94 is right, when the newer commit has been pushed, previous workflows are canceled.
@kimwnasptd You can check example here: https://github.com/kubeflow/katib/actions/runs/3534409566

**Comment by andreyvelich** [comment]
The Charmed seems to be working now: https://github.com/kubeflow/katib/actions/runs/3564777436/jobs/5989685476
@DomFleischmann @johnugeorge @kimwnasptd Please take a look at the changes.

**Comment by tenzen-y** [comment]
@andreyvelich  Thanks for this improvement!

**Pull request merged**